cypress/components/menu/menu.cy.tsx

```diff
@@ -819,44 +422,41 @@ context("Testing Menu component", () => {
       );
 
       submenu().eq(positionOfElement("first")).trigger("mouseover");
-      submenu()
-        .eq(positionOfElement("first"))
-        .then(($menu) => {
-          submenuBlock().should("have.css", "width", `${$menu.width()}px`);
-        });
+      submenu().eq(positionOfElement("first")).as("submenuBlock");
+      cy.get("@submenuBlock").its("0").should("have.css", "width").as("width");
+      cy.get("@width")
+        .then(($el) => parseInt($el.toString()))
+        .should("be.within", 490, 499);
     });
 
     it.each([
-      ["float", 0.3, 409],
-      ["float", 0.6, 819],
-      ["float", 1.0, 1366],
-      ["number", 350, 350],
-      ["number", 900, 900],
-      ["number", 1350, 1350],
-      ["string", "450px", 450],
-      ["string", "675px", 675],
-      ["string", "1200px", 1200],
-    ])(
-      "should verify Menu width is set by width prop when passed as a %s",
-      (type, width, pixels) => {
-        CypressMountWithProviders(<MenuComponent width={width} />);
-
-        menu().then(($el) => {
-          assertCssValueIsApproximately($el, "width", pixels);
-        });
-      }
-    );
+      [0.3, 409],
+      [0.6, 819],
+      [1.0, 1366],
+      [350, 350],
+      [900, 900],
+      [1350, 1350],
+      ["450px", 450],
+      ["675px", 675],
+      ["1200px", 1200],
+    ])("should verify Menu width is set by width prop", (width, pixels) => {
```

Comment:
Again this is up for debate, I just personally think if this test is running 9 times there should be some sort of indicator on the test case, as de-bugging might become difficult. However, @stephenogorman is much more experienced than me, especially in the writing of tests and test descriptions. Happy for you to stick with his recommendation, but whatever we choose its implementation needs to be consistent 👍